### PR TITLE
Queries related to CVE-2022-22965 SpringShell

### PR DESCRIPTION
**Pull request by aprakash13**
Change(s):
   - Queries related to CVE-2022-22965 SpringShell

   Reason for Change(s):
   - New vulnerability found.

   Version Updated:
   -Yes

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

### CODE REVIEW

File: Detections/SecurityAlert/AVSpringShell.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+query: |
+  let SpringShell_threats = dynamic(["SpringShell"]);
+  DeviceInfo
+  | extend DeviceName = tolower(DeviceName)
+  | join ( SecurityAlert
+  | where ProviderName == "MDATP"
+  | extend ThreatFamilyName = tostring(parse_json(ExtendedProperties).ThreatFamilyName)
+  | where ThreatFamilyName in (SpringShell_threats)
+  | extend CompromisedEntity = tolower(CompromisedEntity)
+  ) on $left.DeviceName == $right.CompromisedEntity
```
Comment (shainw) [inline Detections/SecurityAlert/AVSpringShell.yaml:33]: the schema is over 50 columns for this, wondering if we should be doing any kind of summarize or project to reduce output fields?

File: Detections/SecurityAlert/AVSpringShell.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    dataTypes:
+      - SecurityAlert
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1190
```
Comment (shainw) [inline Detections/SecurityAlert/AVSpringShell.yaml:19]: subTechnique?

File: Hunting Queries/AzureDiagnostics/SpringShellExploitationAttempt.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+   malicious JSP file in a location accessible by http requests. Attackers then make requests to the malicious backdoor to run system commands.
+   Refrence: https://www.microsoft.com/security/blog/2022/04/04/springshell-rce-vulnerability-guidance-for-protecting-against-and-detecting-cve-2022-22965/'
+requiredDataConnectors:
+  - connectorId: WAF
+    dataTypes:
+      - AzureDiagnostics
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1190
```
Comment (shainw) [inline Hunting Queries/AzureDiagnostics/SpringShellExploitationAttempt.yaml:16]: sub-Technique?

File: Hunting Queries/AzureDiagnostics/SpringShellExploitationAttempt.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+id: 1bb4a007-7d1d-4506-ada9-222604f54ec6
+name: Possible SpringShell Exploitation Attempt (CVE-2022-22965)
+description: |
+  'This hunting query looks in Azure Web Application Firewall data to find possible SpringShell Exploitation Attempt (CVE-2022-22965).
+   The Spring Framework is one of the most widely used lightweight open-source framework for Java. To exploit the vulnerability attackers can  
+   send a specially crafted query to a web server running the Spring Core framework to change the target of logging facility and create a new 
+   malicious JSP file in a location accessible by http requests. Attackers then make requests to the malicious backdoor to run system commands.
+   Refrence: https://www.microsoft.com/security/blog/2022/04/04/springshell-rce-vulnerability-guidance-for-protecting-against-and-detecting-cve-2022-22965/'
```
Comment (shainw) [inline Hunting Queries/AzureDiagnostics/SpringShellExploitationAttempt.yaml:8]: Reference

**Review by shainw — CHANGES_REQUESTED** [review]
Some items to fix up

File: Detections/SecurityAlert/AVSpringShell.yaml [hunk 5]
```diff
@@ -22,15 +22,17 @@ tags:
...
+  let SpringShell_threats = dynamic(["Trojan:Python/SpringShellExpl", "Exploit:Python/SpringShell", "Backdoor:PHP/Remoteshell.V", "SpringShell"]);
   DeviceInfo
   | extend DeviceName = tolower(DeviceName)
-  | join ( SecurityAlert
+  | join kind=inner ( SecurityAlert
   | where ProviderName == "MDATP"
+  | extend ThreatName = tostring(parse_json(ExtendedProperties).ThreatName)
   | extend ThreatFamilyName = tostring(parse_json(ExtendedProperties).ThreatFamilyName)
-  | where ThreatFamilyName in (SpringShell_threats)
+  | where ThreatName in (SpringShell_threats) or ThreatFamilyName in (SpringShell_threats)
```
Comment (shainw) [inline Detections/SecurityAlert/AVSpringShell.yaml:32]: Should we do in~ just in case the case is different in the match?

**Review by shainw — APPROVED** [review]
One suggestion, maybe not required. I see you updated 1 with subtechnique, the others don't have one?

**Comment by aprakash13** [comment]
> One suggestion, maybe not required. I see you updated 1 with subtechnique, the others don't have one?

The others do not have a subTechnique.

**Pull request merged**